datasetinsights/storage/checkpoint.py

```diff
@@ -5,89 +5,109 @@
 import tempfile
 
 import datasetinsights.constants as const
+from datasetinsights.data.download import download_file
 from datasetinsights.torch_distributed import is_master
 
 from .gcs import GCSClient, gcs_bucket_and_path
 
 logger = logging.getLogger(__name__)
 
 DEFAULT_SUFFIX = "estimator"
-GCS_BASE_STR = "gs://"
 
 
-def checkpoint_file_on_gcs(config):
+class EstimatorCheckpoint:
     """
-    determine whether or not the checkpoint file to load an estimator is on
-    gcs
+    For loading and saving estimator checkpoints
 
     Args:
-        config:
+        estimator_name (str): name of the estimator
+        log_dir (str): log directory
+        distributed (bool): boolean to determine distributed training
+    Attributes:
+        estimator_name (str): name of the estimator
+        log_dir (str): log directory
+        distributed (bool): boolean to determine distributed training
+    """
 
-    Returns:
+    def __init__(self, estimator_name, log_dir, distributed):
+        self.distributed = distributed
+        self._writer = self._create_writer(log_dir, estimator_name)
 
-    """
-    return (
-        "checkpoint_file" in config.keys()
-        and config.checkpoint_file != const.NULL_STRING
-        and config.checkpoint_file[: len(GCS_BASE_STR)] == GCS_BASE_STR
-    )
+    @staticmethod
+    def _create_writer(log_dir, estimator_name):
+        if log_dir.startswith(const.GCS_BASE_STR):
+            writer = GCSEstimatorWriter(log_dir, estimator_name)
+        else:
+            writer = LocalEstimatorWriter(estimator_name)
+
+        return writer
+
+    @staticmethod
+    def _get_loader_from_path(path):
+        if path.startswith((const.HTTP_URL_BASE_STR, const.HTTPS_URL_BASE_STR)):
+            method = load_from_http
+        elif path.startswith(const.GCS_BASE_STR):
+            method = load_from_gcs
+        elif path.startswith("/"):
+            method = load_local
+        else:
+            raise ValueError(
+                f"Given path: {0}, is either invalid or not "
```

Comment:
Use python's f-string and remove instead of `"".format(...)`. Also, this error should be more informative to tell users what. type of paths are supported.